nomad/utils/text.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+
+import re
+import unicodedata
+
+
+_punct_re = re.compile(r'[\t !"#$%&\'()*\-/<=>?@\[\\\]^_`{|},.]+')
+
+
+def slugify(text):
+    """ Returns a ASCII-only slug version of the considered text. """
+    result = []
+    for word in _punct_re.split(text.lower()):
+        word = unicodedata.normalize('NFKD', word).encode('ascii', 'ignore').decode('ascii')
```

Comment:
Currently the project's name can be either manually set in the nomad file or dynamically set using the folder name of the project. eg. if the nomad.yml file is stored in "/path/to/mon projet", "mon projet" will be the name of the project (if the project name is not in the nomad.yml file of course). So we have to enforce that our LXD name contains only character that can be part of a hostname (actually all LXD names must be valid hostnames).

I agree that we should enforce some rules regarding project names and container names (when they are manually set). But this will require some validation logic in order to validate the content of a nomad file prior any operations with containers (my plan was to add validation rules for nomad files with another pull request).

But when the project's name is computed from the project's path, we should slugify this value when using it to create LXD names. :wink: